xds/src/main/java/io/grpc/xds/WeightedRoundRobinLoadBalancer.java

```diff
@@ -390,22 +384,25 @@ static final class StaticStrideScheduler {
         }
       }
 
-      double unscaledMeanWeight = numWeightedChannels > 0
-          ? sumWeight / numWeightedChannels : 1;
-
       // Adjust max value s.t. ratio does not exceed K_MAX_RATIO. This should
       // ensure that we on average do at most K_MAX_RATIO rounds for picks.
-      unscaledMaxWeight = Math.min(unscaledMaxWeight, (float) (K_MAX_RATIO * unscaledMeanWeight));
+      if (numWeightedChannels > 0) {
+        unscaledMeanWeight = sumWeight / numWeightedChannels;
+        unscaledMaxWeight = Math.min(unscaledMaxWeight, (float) (K_MAX_RATIO * unscaledMeanWeight));
+      // Fall back to round robin if all values are non-positives
```

Comment:
Put this on the same line as the else (`} else { // Fall back...`) or first line within the else block.